src/memorylayout.jl

```diff
@@ -606,8 +606,15 @@ rowsupport(::BidiagonalLayout, A, j) =
 colsupport(::AbstractTridiagonalLayout, A, j) = max(minimum(j)-1,1):min(size(A,1),maximum(j)+1)
 rowsupport(::AbstractTridiagonalLayout, A, j) = max(minimum(j)-1,1):min(size(A,2),maximum(j)+1)
 
-colsupport(::SymmetricLayout, A, j) = first(colsupport(symmetricdata(A),j)):last(rowsupport(symmetricdata(A),j))
-rowsupport(::SymmetricLayout, A, j) = colsupport(A, j)
+data(A::Symmetric) = symmetricdata(A)
```

Comment:
`data` here is probably too generic.

Reply:
I will use the existing `parent` for both cases.
I am wondering, why there is no generic function covering `Symmetric` and `Hermitian`. The existence of `symmetricdata` besides `hermitiandata` causes unneeded code duplication.
More generally I wonder, why we nedd `SymmetricLayout` and also `HermitianLayout.

Reply:
`parent` won't work because you don't know if `A` is `Symmetric`: it could be an `Adjoint` or `SubArray`, see

https://github.com/JuliaMatrices/ArrayLayouts.jl/blob/5e0f27556d101d005396725c244562772673d734/test/test_layouts.jl#L121

Reply:
I will go back to the code duplication.

Reply:
OK. We can redesign it later (say, a `symorhermdata`). The issue with just `data` on its own is sometimes an array can have multiple layouts at the same time. But I don't think there is any confusion possible for `SymmetricLayout` and `HermitianLayout`.